### PR TITLE
Egardia: Adding configuration tag according to new docs

### PR DESCRIPTION
**Pull request by jeroenterheerdt**
A request was made in the last doc review to start using the %configuration% tag (PR: https://github.com/home-assistant/home-assistant.github.io/pull/4803). This PR does that. 
One thing is missing (need help!): I did not find a sample of how to specify the keys for the map type in configuration. The component accepts a list of states that can contain lists of codes, like this:
```yaml
report_server_codes:
  arm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  disarm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  armhome: XXXXXXXXXXXXXXXX
  triggered: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  ignore: XXXXXXXXXXXXXXXX
```
not sure how to put this in the %configuration% tag. For now, I left both the old version and the {%configuration%} tag in.

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [X] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [X] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by jeroenterheerdt** [comment]
@balloob @MartinHjelmare here is the new replacement PR for the requested change to the Egardia docs. Not sure if I did it correctly though - for at least one piece I need some help (see description of PR). I closed #4876 since the rebase to `current` did not work. This new PR is based on `current`. Please help figuring out the right way of using the `%configuration%` tag for the list of lists that my component accepts.

**Comment by balloob** [comment]
I don't know how to specify a list of tags either.

**Comment by jeroenterheerdt** [comment]
@balloob if you don't know, who can? I mean, you are the master of this eco system 😀

**Comment by MartinHjelmare** [comment]
Look at the mysensors hub docs. The `gateway` key holds a list of maps of key value pairs. You want a map of key value pairs where the values are lists. So just turn it around.

I don't think you should specify the strings as comma separated. Just make that regular lists of strings. The user can use both ways.

File: source/_components/egardia.markdown [hunk 1]
```diff
@@ -59,9 +65,45 @@ You can change this, however, using the following procedure. This is a more adva
         triggered: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
         ignore: XXXXXXXXXXXXXXXX
     ```
+{% configuration %}
```
Comment (MartinHjelmare) [inline source/_components/egardia.markdown]: Only one of these sections should be written and it should contain all the options.

Comment (jeroenterheerdt) [reply source/_components/egardia.markdown]: ok

**Comment by jeroenterheerdt** [comment]
@MartinHjelmare not sure if I understand what you meant, anyway, here is a first stab at improving. Now only one `{% configuration %}` section remains and I added some headers with `{% linkable_title %}` tags. Not sure if the config is correct right now, because it seems the `mysensors` hub has a slightly different config.

File: source/_components/egardia.markdown [hunk 2]
```diff
@@ -16,40 +17,90 @@ The `egardia` platform enables the ability to control an [Egardia](http://egardi
...
+  required: false
+  type: string
+  default: false
+report_server_port:
+  description:  Port of the Egardia server.
+  required: false
+  type: int
+  default: 52010
+report_server_codes:
+  description: List of codes for the different states.
```
Comment (MartinHjelmare) [inline source/_components/egardia.markdown]: This is a map of list of codes.

File: source/_components/egardia.markdown [hunk 3]
```diff
@@ -16,40 +17,90 @@ The `egardia` platform enables the ability to control an [Egardia](http://egardi
...
+  required: false
+  type: map
+  keys:
+    arm:
+      description: List of codes for the 'arm' state.
+      required: false
+      type: map
```
Comment (MartinHjelmare) [inline source/_components/egardia.markdown]: I think you can specify `list` here, although it's not clearly mentioned in our docs for configuration tag.

Comment (jeroenterheerdt) [reply source/_components/egardia.markdown]: well, it is *not* mentioned, but let me try :)

File: source/_components/egardia.markdown (hunk 2 above)
Comment (jeroenterheerdt) [reply source/_components/egardia.markdown]: ok

**Comment by jeroenterheerdt** [comment]
@MartinHjelmare build failed, not sure why.

**Comment by MartinHjelmare** [comment]
Seems unrelated to this PR:
```
10:00:59 AM: failed during stage 'deploying site': Invalid filename 'getting-started/basic/#remote-access.html'. Deployed filenames cannot contain # or ? characters
```

**Comment by jeroenterheerdt** [comment]
so...?

**Comment by MartinHjelmare** [comment]
I think @balloob fixed some of these build bugs in another PR, but I don't think this one was included. You could try to rebase your branch on current, though, to see if that helps.

**Comment by balloob** [comment]
Yeah the build failed because your base branch is not up to date. Both `HEAD` of `current` and `next` branches are passing.

**Comment by jeroenterheerdt** [comment]
I tried rebasing, the result is above. Let me know if this is OK.

**Comment by MartinHjelmare** [comment]
There's lots of unrelated commits in your branch now.

It's a lot harder to rebase if you're working locally directly on one of the main upstream branches, in this case `current`. Take for habit to always checkout and create a new, with another name, local branch based on one of the main upstream branches when starting your work. Then you can easily rebase the local "feature" branch during the development process

This is outlined here:
https://home-assistant.io/developers/development_submitting/
and here:
https://home-assistant.io/developers/development_catching_up/

**Comment by jeroenterheerdt** [comment]
You are right, I forgot to create a new branch. So what do I do? Delete repo and refork?

**Comment by jeroenterheerdt** [comment]
I tried creating a new fork, but the moment I create a new branch for my changes GitHub says my branch is X commits ahead and Y commits behind home-assistant.next, while I created my new branch based on `current`. Should I be creating my branch based on `next` instead?

**Comment by jeroenterheerdt** [comment]
I created a new PR (replacement of the replacement) here: https://github.com/home-assistant/home-assistant.github.io/pull/4910. Hope it is correct this time.

**Pull request closed without merging**